File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/copy_agent_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class CopyAgentDetails(object):
    """
    Details to copy the agent to target workspace.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new CopyAgentDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param target_display_name:
            The value to assign to the target_display_name property of this CopyAgentDetails.
        :type target_display_name: str

        :param target_description:
            The value to assign to the target_description property of this CopyAgentDetails.
        :type target_description: str

        :param target_path_info:
            The value to assign to the target_path_info property of this CopyAgentDetails.
        :type target_path_info: str

        :param target_workspace_key:
            The value to assign to the target_workspace_key property of this CopyAgentDetails.
        :type target_workspace_key: str

        """
        self.swagger_types = {
            'target_display_name': 'str',
            'target_description': 'str',
            'target_path_info': 'str',
            'target_workspace_key': 'str'
        }

        self.attribute_map = {
            'target_display_name': 'targetDisplayName',
            'target_description': 'targetDescription',
            'target_path_info': 'targetPathInfo',
            'target_workspace_key': 'targetWorkspaceKey'
        }

        self._target_display_name = None
        self._target_description = None
        self._target_path_info = None
        self._target_workspace_key = None

    @property
    def target_display_name(self):
        """
        Gets the target_display_name of this CopyAgentDetails.
        Agent name.


        :return: The target_display_name of this CopyAgentDetails.
        :rtype: str
        """
        return self._target_display_name

    @target_display_name.setter
    def target_display_name(self, target_display_name):
        """
        Sets the target_display_name of this CopyAgentDetails.
        Agent name.


        :param target_display_name: The target_display_name of this CopyAgentDetails.
        :type: str
        """
        self._target_display_name = target_display_name

    @property
    def target_description(self):
        """
        Gets the target_description of this CopyAgentDetails.
        Agent description.


        :return: The target_description of this CopyAgentDetails.
        :rtype: str
        """
        return self._target_description

    @target_description.setter
    def target_description(self, target_description):
        """
        Sets the target_description of this CopyAgentDetails.
        Agent description.


        :param target_description: The target_description of this CopyAgentDetails.
        :type: str
        """
        self._target_description = target_description

    @property
    def target_path_info(self):
        """
        **[Required]** Gets the target_path_info of this CopyAgentDetails.
        Path inside volume where the agent JSON is written.


        :return: The target_path_info of this CopyAgentDetails.
        :rtype: str
        """
        return self._target_path_info

    @target_path_info.setter
    def target_path_info(self, target_path_info):
        """
        Sets the target_path_info of this CopyAgentDetails.
        Path inside volume where the agent JSON is written.


        :param target_path_info: The target_path_info of this CopyAgentDetails.
        :type: str
        """
        self._target_path_info = target_path_info

    @property
    def target_workspace_key(self):
        """
        **[Required]** Gets the target_workspace_key of this CopyAgentDetails.
        Key of the target workspace where the agent is copied.


        :return: The target_workspace_key of this CopyAgentDetails.
        :rtype: str
        """
        return self._target_workspace_key

    @target_workspace_key.setter
    def target_workspace_key(self, target_workspace_key):
        """
        Sets the target_workspace_key of this CopyAgentDetails.
        Key of the target workspace where the agent is copied.


        :param target_workspace_key: The target_workspace_key of this CopyAgentDetails.
        :type: str
        """
        self._target_workspace_key = target_workspace_key

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/CopyAgentDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Details to copy the agent to target workspace.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=CopyAgentDetails.Builder.class)

public final class CopyAgentDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"targetDisplayName", "targetDescription", "targetPathInfo", "targetWorkspaceKey"})
    public CopyAgentDetails(String targetDisplayName, String targetDescription, String targetPathInfo, String targetWorkspaceKey) {
        super();
        this.targetDisplayName = targetDisplayName;
        this.targetDescription = targetDescription;
        this.targetPathInfo = targetPathInfo;
        this.targetWorkspaceKey = targetWorkspaceKey;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Agent name.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("targetDisplayName")
private String targetDisplayName;

        /**
         * Agent name.
         * @param targetDisplayName the value to set
         * @return this builder
         **/
        

public Builder targetDisplayName(String targetDisplayName) {
    this.targetDisplayName = targetDisplayName;
    return this;
}
            /**
     * Agent description.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("targetDescription")
private String targetDescription;

        /**
         * Agent description.
         * @param targetDescription the value to set
         * @return this builder
         **/
        

public Builder targetDescription(String targetDescription) {
    this.targetDescription = targetDescription;
    return this;
}
            /**
     * Path inside volume where the agent JSON is written.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("targetPathInfo")
private String targetPathInfo;

        /**
         * Path inside volume where the agent JSON is written.
         * @param targetPathInfo the value to set
         * @return this builder
         **/
        

public Builder targetPathInfo(String targetPathInfo) {
    this.targetPathInfo = targetPathInfo;
    return this;
}
            /**
     * Key of the target workspace where the agent is copied.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("targetWorkspaceKey")
private String targetWorkspaceKey;

        /**
         * Key of the target workspace where the agent is copied.
         * @param targetWorkspaceKey the value to set
         * @return this builder
         **/
        

public Builder targetWorkspaceKey(String targetWorkspaceKey) {
    this.targetWorkspaceKey = targetWorkspaceKey;
    return this;
}


        public CopyAgentDetails build() {
            CopyAgentDetails model = new CopyAgentDetails(this.targetDisplayName
                , this.targetDescription
                , this.targetPathInfo
                , this.targetWorkspaceKey);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(CopyAgentDetails model) {
                this.targetDisplayName(model.getTargetDisplayName());
    this.targetDescription(model.getTargetDescription());
    this.targetPathInfo(model.getTargetPathInfo());
    this.targetWorkspaceKey(model.getTargetWorkspaceKey());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Agent name.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("targetDisplayName")
    private final String targetDisplayName;

        /**
     * Agent name.
     * @return the value
     **/
    
    public String getTargetDisplayName() {
        return targetDisplayName;
    }


        /**
     * Agent description.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("targetDescription")
    private final String targetDescription;

        /**
     * Agent description.
     * @return the value
     **/
    
    public String getTargetDescription() {
        return targetDescription;
    }


        /**
     * Path inside volume where the agent JSON is written.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("targetPathInfo")
    private final String targetPathInfo;

        /**
     * Path inside volume where the agent JSON is written.
     * @return the value
     **/
    
    public String getTargetPathInfo() {
        return targetPathInfo;
    }


        /**
     * Key of the target workspace where the agent is copied.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("targetWorkspaceKey")
    private final String targetWorkspaceKey;

        /**
     * Key of the target workspace where the agent is copied.
     * @return the value
     **/
    
    public String getTargetWorkspaceKey() {
        return targetWorkspaceKey;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("CopyAgentDetails(");
        sb.append("targetDisplayName=").append(String.valueOf(this.targetDisplayName));
        sb.append(", targetDescription=").append(String.valueOf(this.targetDescription));
        sb.append(", targetPathInfo=").append(String.valueOf(this.targetPathInfo));
        sb.append(", targetWorkspaceKey=").append(String.valueOf(this.targetWorkspaceKey));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof CopyAgentDetails)) {
            return false;
        }

        CopyAgentDetails other = (CopyAgentDetails) o;
        return java.util.Objects.equals(this.targetDisplayName, other.targetDisplayName) &&
            java.util.Objects.equals(this.targetDescription, other.targetDescription) &&
            java.util.Objects.equals(this.targetPathInfo, other.targetPathInfo) &&
            java.util.Objects.equals(this.targetWorkspaceKey, other.targetWorkspaceKey);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.targetDisplayName == null ? 43 : this.targetDisplayName.hashCode());
        result = (result * PRIME) + (this.targetDescription == null ? 43 : this.targetDescription.hashCode());
        result = (result * PRIME) + (this.targetPathInfo == null ? 43 : this.targetPathInfo.hashCode());
        result = (result * PRIME) + (this.targetWorkspaceKey == null ? 43 : this.targetWorkspaceKey.hashCode());
        return result;
    }


}
